### DISCUSSION TITLE
Version 2024.07

### DISCUSSION THREAD

**Discussion by kenta2** [B-Lang-org/bsc#735] [Releases]
This version includes the `BuildList` library, syntactical improvements to included and generated C and C++ code, improvements to documentation, and other general improvements.
For a list of changes since the previous release, see the [release notes](https://github.com/B-Lang-org/bsc/blob/2024.07/release/ReleaseNotes.adoc).

Documentation and pre-built tar-files are attached below. As verification of integrity, the sha256sum hashes of the tar-files are
```
d3a0d48fd10f138b5837a0aa9466ad296b257ea984a8a784856672dcc575d768  bsc-2024.07-debian-10.13.tar.gz
6cf22543c027af32e6c6ef7bcb25ceb3610b184be3b3c7132bf645d132ee208d  bsc-2024.07-debian-11.11.tar.gz
e8f1787958185304f560c467f2226fa60e529bebea3841aa99503a9ec43b5ec4  bsc-2024.07-debian-12.7.tar.gz
96dab34d219ac002ec83eca10b501295e586a095f47f25a1e39eedd8bec6ece0  bsc-2024.07-macos-12.tar.gz
61d462aa9ca865edba1fff6de205b2b16c5b8ec1b4444862be01d6eb4321e1c3  bsc-2024.07-macos-13.tar.gz
a084a385fc4ffd03a4bc65d5d14064ccb2ecf537ca38dfc017f5267bd65d754b  bsc-2024.07-macos-14.tar.gz
998ff092f360ecb8ede40ac001b844dd2b33d0bb36ec78b1d925e8ada4f24672  bsc-2024.07-rockylinux-8.10.tar.gz
7665848e11ac255dd02f5911d6cd7219efe7b2cf1bdc660c7cdaa201e2355604  bsc-2024.07-rockylinux-9.4.tar.gz
7a96fd7d7348b45820f2a253b3ffa6f00a175de6a959f0340e1b2b66ba5acbc7  bsc-2024.07-ubuntu-18.04.tar.gz
629af50d6e55715e3de72596e5518f2e03b2deb1cf1b32127aa2608cc82a24d6  bsc-2024.07-ubuntu-20.04.tar.gz
a5d48eb5242955ef1b8aad3cc4051e282ec8de4a1f9545413ab8fa2d064b6aa1  bsc-2024.07-ubuntu-22.04.tar.gz
0f35d54937cd776159d9e053aff57b9a5d2d642e0e751c06aac3028678ec7e37  bsc-2024.07-ubuntu-24.04.tar.gz
a7211d089be68303983cc644b70edaae8efab529ff63fd8670a4f20119888781  yices-src-for-bsc-2024.07.tar.gz
```

Debian 10 is "Buster", Debian 11 is "Bullseye", and Debian 12 is "Bookworm", the current Debian Stable.  Rocky Linux 8 is compatible with RHEL 8, and Rocky Linux 9 is compatible with RHEL 9. Ubuntu 18.04 is "Bionic Beaver", Ubuntu 20.04 is "Focal Fossa", and Ubuntu 22.04 is "Jammy Jellyfish".  The ubuntu-24.04 tar-file was built on a pre-release of Ubuntu 24.04 "Noble Numbat".  MacOS 11 is "Big Sur", macOS 12 is "Monterey", macOS 13 is "Ventura", macOS 14 is "Sonoma". All releases are built for x86_64, except macOS 14 "Sonoma" .  The release for macOS 14 "Sonoma" is built for ARM64 (Apple Silicon).

The "Source code" attached below is automatically created by GitHub and does not include the source for submodules. We have separately attached the source code for the Yices submodule, that can be unpacked into the BSC source, to allow building from this snapshot.

<hr /><em>This discussion was created from the release <a href='https://github.com/B-Lang-org/bsc/releases/tag/2024.07'>Version 2024.07</a>.</em>

**Comment by rsnikhil** [comment]
I'm pleased to report that:

    'bsc-2024.07-macos-14.tar.gz' (arm64 executable)

seems to work fine on my arm64 laptop which is running an *earlier* MacOS:

    MacBook Air M2 2022 (arm64), MacOS Ventura 13.6.9

I've successfully compiled and run my Fife and Drum RISC-V CPUs,
in Bluesim and Verilator sim, with imported C code.

As with the earlier 'bsc-2024.01-macos-13.tar.gz' (x86_64 executable)
which I have been using so far, one gets four MacOS security warnings
on first run.  The first one is:

    "bsc" cannot be opened because the developer cannot be verified."
At this point, opening

    Settings->'Privacy & Security'

one finds a new entry for 'bsc', with a button "Allow anyway"
Clicking this button allows proceeding further.

There are three more security warnings; each can be overcome in the
same way:

    libstp.dylib
    libyices.2.dylib
    bluetcl

after which everything proceeds smoothly.
